openslides_backend/presenter/search_users_by_name_email.py

```diff
@@ -0,0 +1,143 @@
+
+import fastjsonschema
+
+from ..action.actions.user.user_scope_permission_check_mixin import UserScope
+from ..permissions.management_levels import (
+    CommitteeManagementLevel,
+    OrganizationManagementLevel,
+)
+from ..permissions.permission_helper import (
+    has_committee_management_level,
+    has_organization_management_level,
+    has_perm,
+)
+from ..permissions.permissions import Permissions
+from ..shared.exceptions import MissingPermission, PresenterException
+from ..shared.filters import FilterOperator, Or
+from ..shared.patterns import Collection, FullQualifiedId
+from ..shared.schema import schema_version
+from .base import BasePresenter
+from .presenter import register_presenter
+
+search_users_by_name_email_schema = fastjsonschema.compile(
+    {
+        "$schema": schema_version,
+        "type": "object",
+        "title": "search_users_by_name_email",
+        "description": "get lists of id, first-, last-name and email for tuples of exact (username, emails)-tuples.",
+        "properties": {
+            "permission_scope": {
+                "type": "array",
+                "prefixItems": [
+                    {
+                        "enum": [1, 2, 3]
+                    },  # Permission scopes Meeting, Committee, Organization
+                    {"type": "number"},  # Id of object, 1 for organization
+                ],
+            },
+            "search": {
+                "type": "array",
+                "minItems": 1,
+                "items": {
+                    "type": "array",
+                    "minItems": 2,
+                    "maxItems": 2,
+                    "prefixItems": [
+                        {"type": "string"},  # username
+                        {"type": "string"},  # email-adress
+                    ],
+                },
+            },
+        },
+        "required": ["permission_scope"],
+        "additionalProperties": False,
+    }
+)
+
+
+@register_presenter("search_users_by_name_email")
+class SearchUsersByNameEmail(BasePresenter):
+    """
+    Collects users with exect usernames or exact emails.
+    """
+
+    schema = search_users_by_name_email_schema
+
+    def get_result(self) -> Any:
+        self.check_permissions(self.data["permission_scope"])
+        result: Dict[str, List[Tuple[int, str, str, str]]] = {}
+        for search in self.data["search"]:
+            if search[0] and search[1]:
+                filter_ = Or(
+                    FilterOperator("username", "=", search[0]),
+                    FilterOperator("email", "=", search[1]),
+                )
+            elif search[0]:
+                filter_ = FilterOperator("username", "=", search[0])
+            elif search[1]:
+                filter_ = FilterOperator("email", "=", search[1])
+            else:
+                continue
+            instances = self.datastore.filter(
+                Collection("user"), filter_, ["id", "first_name", "last_name", "email"]
```

Comment:
looking at this, this will probably lead to performance issues when there are millions of users, since there is no index to use...